sklearn/manifold/_mds.py

```diff
@@ -675,32 +746,82 @@ def fit_transform(self, X, y=None, init=None):
 
         if self.n_init == "warn":
             warnings.warn(
-                "The default value of `n_init` will change from 4 to 1 in 1.9.",
+                "The default value of `n_init` will change from 4 to 1 in 1.9. "
+                "To suppress this warning, provide some value of `n_init`.",
                 FutureWarning,
             )
             self._n_init = 4
         else:
             self._n_init = self.n_init
 
+        if self.init == "warn":
+            warnings.warn(
+                "The default value of `init` will change from 'random' to "
+                "'classical_mds' in 1.10. To suppress this warning, provide "
+                "some value of `init`.",
+                FutureWarning,
+            )
+            self._init = "random"
+        else:
+            self._init = self.init
+
+        if self.dissimilarity != "deprecated":
```

Comment:
we should also raise a `ValueError` if `dissimilarity` and `metric` (non-boolean) are both provided.

Reply:
Okay, adding this now, but I have to allow providing `dissimilarity` when `metric="euclidean" because this is default value of `metric`. I guess it's fine.